pymaster/field.py

```diff
@@ -658,3 +659,240 @@ def get_ell_sampling(self):
         """
         ells = lib.get_ell_sampling_flat(self.fl, int(self.fl.fs.n_ell))
         return ells
+
+
+class NmtFieldCatalog(NmtField):
+    """
+    """
+    def __init__(self, positions, weights, field, spin=None, beam=None,
+                 field_is_weighted=False, lmax=-1, lmax_mask=-1):
+        # 0. Preliminary initializations
+        if ut.HAVE_DUCC:
+            self.sht_calculator = 'ducc'
+        else:
+            raise ValueError("DUCC is needed, but currently not installed.")
+
+        # These first attributes are compulsory for all fields
+        self.lite = True
+        self.mask = None
+        self.beam = None
+        self.n_iter = None
+        self.n_iter_mask = None
+        self.pure_e = False
+        self.pure_b = False
+        self.Nw = 0.
+        self.Ngamma = 0.
```

Comment:
shall we call this `Nfield`, `field_noise` or something like that? Since this is not meant to be specific of cosmic shear.